include/dyad/c_api/dyad_c.h

```diff
@@ -0,0 +1,34 @@
+#ifndef DYAD_C_API_DYAD_C_H
+#define DYAD_C_API_DYAD_C_H
+
+#if defined(DYAD_HAS_CONFIG)
+#include <dyad/dyad_config.hpp>
+#else
+#error "no config"
+#endif
+
+#include <dyad/core/dyad_ctx.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct dyad_ctx;
+typedef struct dyad_ctx dyad_ctx_t;
+
+__attribute__ ((__visibility__ ("default"))) int dyad_c_init (dyad_ctx_t** ctx);
```

Comment:
I was reading somewhere that the visibility attribute differs between GCC, Clang, and cray compilers. I think we should have a macro for this so that it's easier to switch to in the future.

Reply:
I didn't realize that. I would use the macros we have in `dyad_rc.h` if I had known. What I can do is make `dyad_rc.h` a public header file and use the macro here.